### ISSUE TITLE
[api-extractor] does not recognize types when exported via `export = ....`

### ISSUE THREAD

**Issue by phryneas** [microsoft/rushstack#1608]
## Please prefix the issue title with the project name i.e. [rush], [api-extractor] etc. ##

**Is this a feature or a bug?**

- [ ] Feature
- [X] Bug

**Please describe the actual behavior.**

In out repo, we don't `export default X`, but `export = X` (I guess due to legacy reasons). Api-Extractor does not seem to pick that up at all. Do you have any ideas on how to make that work?

**If the issue is a bug, how can we reproduce it? Please provide detailed steps and include a GitHub branch if applicable. Your issue will get resolved faster if you can make it easy to investigate.**

I have a repo here - just run `yarn install; yarn build;  yarn api-extractor run --local --verbose --diagnostics`  - it visits all relevant files, but ouput is empty. Changing the export to a default export makes it work.
https://github.com/phryneas/fork-ts-checker-webpack-plugin/tree/api-extractor

**What is the expected behavior?**

I expect content to be in my reports.

**If this is a bug, please provide the tool version, Node.js version, and OS.**

* **Tool:** api-extractor
* **Tool Version:** 7.5.2
* **Node Version:** 10.15.3
  * **Is this a LTS version?** yup
  * **Have you tested on a LTS version?**  
* **OS:** linux

**Comment by octogonz** [comment]
`export = X` is an obsolete style of exports.  It is useful when writing typings for legacy JavaScript packages, but it should NOT be used by modern TypeScript packages.

Could you clarify which API Extractor features you're using?
1. API review
2. .d.ts rollups
3. generating API documentation

For (1) and (3) an easy workaround would be to create a separate entry point (e.g. `src/index-api.ts`) that exports your API using a modern `export`.  Then you can point API Extrator's [mainEntryPointFilePath](https://api-extractor.com/pages/commands/config_file/#mainentrypointfilepath) at `lib/index-api.d.ts`.

For (2), we would need to implement support for this.  It's possibly not a whole lot of work; however, I wouldn't give it a high priority, since it's a deprecated use case.  If someone in the community wants to contribute a PR, we would certainly accept it, and I'd be happy to help with design/guidance.

**Comment by phryneas** [comment]
Yeah, unfortunately it's (2). I haven't looked at api-extractor yet - if you can point me in the right direction I might try and do that PR myself.
